presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -856,10 +856,18 @@ protected Object visitNotExpression(NotExpression node, Object context)
         @Override
         protected Object visitLogicalBinaryExpression(LogicalBinaryExpression node, Object context)
         {
+            LogicalBinaryExpression.Type type = node.getType();
             Object left = process(node.getLeft(), context);
+            if (type == LogicalBinaryExpression.Type.AND && Boolean.FALSE.equals(left)) {
+                return false;
+            }
+            if (type == LogicalBinaryExpression.Type.OR && Boolean.TRUE.equals(left)) {
```

Comment:
I would prefer to put this logic inside the switch below (also moving the `Object right = process(node.getRight(), context);` there, after short-circuiting is applied)

Reply:
No... I would prefer to leave it here. `right` is used outside of the `switch`, and needs to be calculated for both `OR` and `AND cases.